testing/fpgaconf/test_fpgaconf_c.cpp

```diff
@@ -288,16 +288,16 @@ TEST_P(fpgaconf_c_p, get_bits_err2) {
  * @test       get_bits_err3
  * @brief      Test: get_bitstream_ifc_id
  * @details    When get_bitstream_ifc_id is passed a bitstream buffer,<br>
- *             and that buffer has a json data length field of that is invalid,
- *             then the fn returns FPGA_EXCEPTION.<br>
+ *             and that buffer has a json data length field of that is valid,
+ *             then the fn returns FPGA_OK.<br>
  */
 TEST_P(fpgaconf_c_p, get_bits_err3) {
   struct bitstream_info info;
   EXPECT_EQ(read_bitstream(tmp_gbs_, &info), 0);
 
   fpga_guid guid;
   *(uint32_t *) (info.data + 16) = 65535;
-  EXPECT_EQ(get_bitstream_ifc_id(info.data, &guid), FPGA_EXCEPTION);
+  EXPECT_EQ(get_bitstream_ifc_id(info.data, &guid), FPGA_OK);
```

Comment:
I investigated this a bit. When running this test a thousand times, the memcpy_s call fails in some and succeeds in some. I'm guessing that since we are trying to create an illegal read at that memory region but have no guarantee that the process won't be able to access that point, it acts like this. So the test is invalid. Yes? No?

Reply:
Thanks for pointing this out, Isaac!  Yes, this test was intended to fail.  We need to figure out why it is passing some of the time.

Reply:
fixed
if  metadata size is grater then bit stream size returns FPGA_EXCEPTION